src/main/java/org/example/Main.java

```diff
@@ -122,338 +163,205 @@ public boolean isHappy() {
         welcome();
         Scanner scanner = new Scanner(System.in);
-        while (true) { // I suggest splitting this code in small static methods
+        // I suggest splitting this code in small static methods
+        // TODO how to split them in methods
+        // if first method parse input, then I need to return startNumber, count and users, in this case do I need to
+        // create a structure for it
+        // what is the meaning of TODO?
+        // Do I need to create a jira for it?
+        while (true) {
             System.out.println("Enter a request:");
-            String x = scanner.nextLine();
-            String[] result = x.split(" ");
-            if (result.length == 1) {
-                if (x.equals("0")) {
-                    System.out.println("Goodbye!");
-                    break;
-                } else if (Long.parseLong(x) >= 1) {
-                    MyNumber num = new MyNumber(x);
-                    num.printInfo();
-                } else {
-                    System.out.println("The first parameter should be a natural number or zero.");
+            String inputStr = scanner.nextLine();
+
+            if (inputStr.equals("0")) {
+                System.out.println("Goodbye!");
+                break;
+            }
+            String[] input = inputStr.split(" ");
+            long startNumber = parseNum(input[0]);
+            if (startNumber < 1) {
+                System.out.println("The first parameter should be a natural number or zero.");
+                continue;
+            }
+
+            if (input.length == 1) {
+                MyNumber num = new MyNumber(inputStr);
+                num.printInfo();
+                continue;
+            }
+
+            long count = parseNum(input[1]);
+            if (count < 1) {
+                System.out.println("The second parameter should be a natural number.");
+                continue;
+            }
+
+            // TODO how to do it?
+            Collection<String> users = new
+                    HashSet<>(Arrays.asList(input).subList(2, input.length).stream().map((x) -> x.toUpperCase()).toList());
+
+            if (hasInvalidProperty(users)) {
+                continue;
+            }
+
+            if (hasExclusiveProperty(users)) {
+                continue;
+            }
+
+            processManyArguments(users, startNumber, count);
+        }
+    }
+
+    private static void processManyArguments(Collection<String> users, long startNumber, long count) {
+        System.out.println();
+        long cur = startNumber;
+        while (count > 0) {
+            MyNumber num = new MyNumber(Long.toString(cur++));
+            if (users.size() > 0) {
+                boolean toSkip = false;
+                // TODO use var or proper name
+                for (var param : MyNumber.PARAMS.values()) {
+                    toSkip |= check(users, param.name(), (x) -> !(x.params.contains(param)), num);
```

Comment:
Instead of exposing params field in MyNumber class it's better to create a method containsParam(String) in MyNumber and call it here.